### ISSUE TITLE
Consistent handling of label encoding and logit squashing

### ISSUE THREAD

**Issue by tordks** [tordks/image_classification#14]
The way encoding of labels and squashing of logits are handled is not consistent. Tidy up by:
* Make the dataset/datamodule handle ohe encoding
* Make the LightningModule handle the squashing
